### ISSUE TITLE
NuGet restore should log package locations on disk

### ISSUE THREAD

**Issue by KirillOsenkov** [NuGet/Home#7691]
NuGet.exe: 4.9.2.5706

NuGet restore should log the full paths to all packages on disk. If a package is downloaded, it should print the destination location.

Right now it just says `Installing Foo.Bar 1.0.0.25.` but it doesn't say **where**, which is the most crucial piece of information.

If it finds that a package is available in the package cache, it should print a message (Found package at location...)

This would greatly simplify investigations of difficult build issues on CI etc.

**Comment by jainaashish** [comment]
Some of the information you asked are already there in current logs like when a package being downloaded, it shows GET http logs or it only shows `installing xxx` for missing packages, etc.. but I agree these are not intutive and hard to parse through. So we can certainly improve these logs or messages and there are already handful of bugs which asks to improve the current loggings. So when we try improving the logs, we'll look at all these bugs to make it a better experience.

**Comment by KirillOsenkov** [comment]
The log shows where the packages are downloaded from, but I don't think they shows where the files are being saved to.

Please note that this information is crucial for difficult build problem investigations and we should prioritize this if at all possible.

**Comment by jainaashish** [comment]
They always go to `globalPackagesFolder` which if not overwritten then will be `%userprofile%/.nuget/packages`. But agreed we can either add this path with individual packages or just write an additional line indicating what was the global packages folder where we restored all the packages.

**Comment by KirillOsenkov** [comment]
activity has occurred
